### PR TITLE
[MM-25408] Improve the grammar on Account Creation Screen

### PR DESCRIPTION
**Pull request by y-martinez**
#### Summary

This PR changes the error message that appears in the Account Creation Screen if the email address you entered does not belong to the accepted domain. This is the result after the change (see screenshot)

mattermost

#### Ticket Link

Fix: #15927 
JIRA: https://mattermost.atlassian.net/browse/MM-25408

```release-note
NONE
```

### CODE REVIEW

**Comment by ashishbhate** [comment]
A few things to check:

1. That error is used in other places too. It might not be correct to change the error message in those places, especially if those errors return multiple emails (where the plural form may be correct). 

2. The webapp has end to end (e2e) cypress tests that will need to be updated as well:

- https://github.com/mattermost/mattermost-webapp/blob/master/e2e/cypress/integration/enterprise/guest_accounts/guest_invitation_ui_spec.js#L324

- https://github.com/mattermost/mattermost-webapp/blob/master/e2e/cypress/integration/team_settings/invite_user_to_closed_team_spec.js#L40

- https://github.com/mattermost/mattermost-webapp/blob/master/e2e/cypress/integration/team_settings/join_closed_team_with_not_allowed_email_spec.js#L68

**Comment by y-martinez** [comment]
> A few things to check:
> 
> 1. That error is used in other places too. It might not be correct to change the error message in those places, especially if those errors return multiple emails (where the plural form may be correct).
> 2. The webapp has end to end (e2e) cypress tests that will need to be updated as well:
> 
> * https://github.com/mattermost/mattermost-webapp/blob/master/e2e/cypress/integration/enterprise/guest_accounts/guest_invitation_ui_spec.js#L324
> * https://github.com/mattermost/mattermost-webapp/blob/master/e2e/cypress/integration/team_settings/invite_user_to_closed_team_spec.js#L40
> * https://github.com/mattermost/mattermost-webapp/blob/master/e2e/cypress/integration/team_settings/join_closed_team_with_not_allowed_email_spec.js#L68

1. I thought in the case that there was more than one domain (just the case of the attached screenshot), however, I did not know that the error message was used in other sites, it is my first time contributing and the code can be overwhelming the first time you see it, I'm going to check which other sites it is used on, but the solutions that come to mind are:

- Keep two different messages to handle these cases in singular and plural (maybe the best option)
- Write a message in such a way that it is correct for both singular and plural. Here if I need a little help because my English is not my first language and I could make spelling mistakes.

What do you think?

2. You are right and I actually made these changes, but I didn't know if I had to do a PR on each repository and link it to the same issue.

**Review by reflog — CHANGES_REQUESTED** [review]
I'm not sure this change is improving the grammar here (though I might be wrong, pinging @justinegeffen ). There could be several addresses, thus the use of plural

**Comment by ashishbhate** [comment]
I checked on the team invite screen, its good there:

image

**Comment by y-martinez** [comment]
@ashishbhate @reflog  You are right, I already saw what the problem is, but in this case what I thought is to have two types of messages, one in the plural and the other in the singular. What do you think?

**Comment by ashishbhate** [comment]
Yep, I think depending on where the error messaged is used we might want to use a different message. I think the i18n strings support variations for singular and plural forms (`one` and `other` keys).

I'll defer to @justinegeffen and @esethna

**Comment by justinegeffen** [comment]
> Yep, I think depending on where the error messaged is used we might want to use a different message. I think the i18n strings support variations for singular and plural forms (`one` and `other` keys).
> 
> I'll defer to @justinegeffen and @esethna

Thanks, @ashishbhate! AFAIK, the i18n strings do support variation for singular and plural; I've definitely made a similar change in the past. 

WRT this specific change, grammatically it is correct but functionally it may be best to retain the original message and then add this one to cover singular and plural. However, this does require consultation with the various maintainers of the translations to confirm when the change can be made across the board. 

Hope this helps somewhat!

**Comment by justinegeffen** [comment]
I'm 5/5 for having a singular and a plural string. @wget, is it best to post this in the i18n channel or just raise a Jira issue for the rest of the strings to be translated?

**Comment by y-martinez** [comment]
> I'm 5/5 for having a singular and a plural string. @wget, is it best to post this in the i18n channel or just raise a Jira issue for the rest of the strings to be translated?

So what can I do while making that decision?

**Comment by wget** [comment]
@justinegeffen It's basically up to you ^^. But when publishing an issue on Jira and removing it from GitHub, it basically becomes invisible for the community used to use GitHub. So if it were to be moved, just do it with care with proper links to keep the discussion going :)

From my experience, we have more developer feedback on GitHub instead of leading a discussion on the Mattermost community instance. But that's just my view.

For this particular case, we have the issue a bit everywhere in the source code. This issue is part of a larger issue I have been describing from the start: developers shoul**n't** tokenize strings :) e.g. [PR#4929](https://github.com/mattermost/mattermost-webapp/pull/4929)

**Comment by srkgupta** [comment]
> 2\. The webapp has end to end (e2e) cypress tests that will need to be updated as well:
> 
> * https://github.com/mattermost/mattermost-webapp/blob/master/e2e/cypress/integration/enterprise/guest_accounts/guest_invitation_ui_spec.js#L324
> * https://github.com/mattermost/mattermost-webapp/blob/master/e2e/cypress/integration/team_settings/invite_user_to_closed_team_spec.js#L40
> * https://github.com/mattermost/mattermost-webapp/blob/master/e2e/cypress/integration/team_settings/join_closed_team_with_not_allowed_email_spec.js#L68

HI @y-martinez 

Any updates on this change requested by @ashishbhate. Can you please create a separate PR on mattermost-webapp with this change and then merge the 2 PRs together when they are finally approved. The rest of the changes in this PR looks good.

**Comment by y-martinez** [comment]
Hi! @srkgupta I don't make these change because I didn't know if I need to have the approval of the new message in plural, that is, all the discussion that @reflog @justinegeffen @wget had, so for that reason I asked that in the meantime what could I do :(

**Comment by srkgupta** [comment]
Oops sorry to jump in so early @y-martinez. I will let you wait once the final decision is made on this. Thanks.

**Comment by jfrerich** [comment]
@reflog @justinegeffen @wget, could one of you help to ensure @y-martinez has direction on next steps?  Thanks!

**Comment by y-martinez** [comment]
There are any updates of this :-/

**Comment by justinegeffen** [comment]
> There are any updates of this :-/

Sorry for the delay, @y-martinez! I've assigned myself to this PR and will review it next week. :)

File: i18n/en.json
Comment (justinegeffen) [inline i18n/en.json:2459]:
```suggestion
    "translation": "The email address you entered does not belong to an accepted domain: {{.Addresses}}. Please contact your System Admin for details."
```

File: i18n/nl.json
Comment (justinegeffen) [inline i18n/nl.json:1755]:
```suggestion
    "translation": "The email address you entered does not belong to an accepted domain: {{.Addresses}}. Please contact your System Admin for details."
```

**Comment by justinegeffen** [comment]
I've re-reviewed this and I think it would be best to have a singular string and a plural string - so basically retain the previous one and add the new one. In terms of i18n work, I recommend joining the [i18n - Localization](https://community-daily.mattermost.com/core/channels/localizatio) channel and getting feedback from the maintainers. But as far as the wording goes, it looks good to me.

**Comment by justinegeffen** [comment]
So basically you'd have:

"The email address you entered does not belong to an accepted domain: {{.Addresses}}. Please contact your System Admin for details."

and

"The email addresess you entered do not belong to an accepted domain: {{.Addresses}}. Please contact your System Admin for details."

**Comment by esethna** [comment]
@y-martinez are you still working on this?

**Comment by justinegeffen** [comment]
Following up on this to check whether it's still in flight. :)

**Comment by isacikgoz** [comment]
Hi @y-martinez, big thanks for your contribution! There hasn't been recent activity on this PR so I've marked it as inactive. Following our [inactive contribution process](https://developers.mattermost.com/contribute/getting-started/inactive-contributions/), the PR is now closed due to 30 days or more of inactivity.

However, if you're still interested working on the changes, let us know! Note that we [merged multiple repositories into one monorepo](https://github.com/mattermost/mattermost-server/issues/22420), which may require you to resubmit the changes. 

If you have any questions, don't hesitate to let us know, we're happy to help.

**Pull request closed without merging**
